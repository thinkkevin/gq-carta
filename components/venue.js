import React, { useContext } from "react";
import Link from "next/link";
import { venueUrl } from "./utils/url";
import { venueLocation } from "./utils/location";
import { UserLocationContext } from "./user-location";
import HighLightText from "./high-light";

import venueStyle from "./venue.module.css";

export function VenueItem(props) {
  let userLocation = useContext(UserLocationContext);

  return (
    <div className={venueStyle.item}>
      <div className={venueStyle.name}>
        <Link href={`${venueUrl(props)}`}>
          <a>
            <HighLightText value={props.name} />
          </a>
        </Link>
      </div>
      <div className={venueStyle.address}>
        {venueLocation(props.location, userLocation)}
      </div>
    </div>
  );
}

export function VenueList({ data = [] }) {
  return (
    <div className={venueStyle.list}>
      {data.map((venue) => {
        return <VenueItem key={venue.id} {...venue} />;
      })}
    </div>
  );
}
