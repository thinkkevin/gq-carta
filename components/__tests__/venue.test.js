import { shallow } from "enzyme";
import { VenueList, VenueItem } from "../venue";
const venues = require("./venue-data.json");

describe("Test VenueList Display", () => {
  it("validate VenueItem component", () => {
    let venue = venues.response.venues[0];
    let comp = shallow(<VenueItem {...venue} />);

    expect(comp.exists(".item")).toEqual(true);
    expect(comp.exists(".name")).toEqual(true);
    expect(comp.exists(".address")).toEqual(true);
    expect(comp.exists(".category")).toEqual(false);

    expect(comp.html()).toMatchSnapshot();
  });

  it("validate VenueList component", () => {
    let comp = shallow(<VenueList data={venues.response.venues} />);
    expect(comp.exists(".list")).toEqual(true);
    expect(comp.find(".list").children().length).toEqual(3); // has venue item as direct children
    expect(comp.find("VenueItem").length).toEqual(3); // has VenueItem as children
  });
});
