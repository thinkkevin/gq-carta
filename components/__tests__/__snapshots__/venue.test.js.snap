// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VenueList Display validate VenueItem component 1`] = `"<div class=\\"item\\"><div class=\\"name\\"><a href=\\"/v/bt-dim-sum/4e07ae1c14959022c8986dbf\\"><span>B&amp;T Dim sum</span></a></div><div class=\\"address\\">36689 Fremont Blvd, Fremont, CA, United States</div></div>"`;
